### PR TITLE
Created problem_44 in project_euler

### PR DESCRIPTION
**Pull request by Kush1101**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by Kush1101** [comment]
@cclauss, Please review it.

**Comment by Kush1101** [comment]
@cclauss Please review it.

File: project_euler/problem_44/sol1.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    Returns True if n is pentagonal, False otherwise.
+    >>> is_pentagonal(330)
+    True
+    >>> is_pentagonal(7683)
+    False
+    >>> is_pentagonal(2380)
+    True
+    """
+    root = (1 + 24 * n) ** 0.5
+    return True if ((1 + root) / 6) % 1 == 0 else False
```
Comment (cclauss) [inline project_euler/problem_44/sol1.py:24]:
```suggestion
    return ((1 + root) / 6) % 1 == 0
```

**Comment by Kush1101** [comment]
@cclauss I made the change.

**Comment by Kush1101** [comment]
@cclauss I made the changes. Please review again.

File: project_euler/problem_44/sol1.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    """
+    pentagonal_nums = [(i * (3 * i - 1)) // 2 for i in range(1, limit)]
+    for i in range(len(pentagonal_nums)):
+        for j in range(i, len(pentagonal_nums)):
+            a, b = (
+                (pentagonal_nums[i] + pentagonal_nums[j]),
+                (pentagonal_nums[j] - pentagonal_nums[i]),
+            )
+            if is_pentagonal(a) and is_pentagonal(b):
+                return pentagonal_nums[j] - pentagonal_nums[i]
```
Comment (cclauss) [inline project_euler/problem_44/sol1.py:42]: Whenever you see `range(len())`, try to use https://docs.python.org/3/library/functions.html#enumerate instead.
```suggestion
    for i, pentagonal_i in enumerate(pentagonal_nums):
        for j in range(i, len(pentagonal_nums)):
            pentagonal_j = pentagonal_nums[j]
            a = pentagonal_i + pentagonal_j
            b = pentagonal_j - pentagonal_i
            if is_pentagonal(a) and is_pentagonal(b):
                return b
```

File: project_euler/problem_44/sol1.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    True
+    """
+    root = (1 + 24 * n) ** 0.5
+    return ((1 + root) / 6) % 1 == 0
+
+
+def compute_num(limit: int = 5000) -> int:
+    """
+    Returns the minimum difference of two pentagonal numbers P1 and P2 such that
+    P1 + P2 is pentagonal and P1-P2 is pentagonal.
```
Comment (cclauss) [inline project_euler/problem_44/sol1.py:30]:
```suggestion
    P1 + P2 is pentagonal and P2 - P1 is pentagonal.
```

**Comment by Kush1101** [comment]
@cclauss Can this be merged?

**Review by cclauss — APPROVED** [review]

**Pull request merged**